File: leetCode/fukun/701-/lc_925.js
/**
 * @param {string} name
 * @param {string} typed
 * @return {boolean}
 */
var isLongPressedName = function(name, typed) {
    let curr1=0,curr2=0;
    while(curr1<name.length||curr2<typed.length){
        if(name[curr1]!==typed[curr2])return false;
        if(typed[curr2]!==typed[curr2+1]||name[curr1]===name[curr1+1])curr1++;
        curr2++;
    }
    return true;
};

File: leetCode/linovce/lc_925.java
class Solution {
    public boolean isLongPressedName(String name, String typed) {
        int j=0;
        int i=0;
        for(;i<name.length()&&j<typed.length();){
            if(name.charAt(i)==typed.charAt(j)){
                i++;
                j++;
            }else if(i>0&&name.charAt(i-1)==typed.charAt(j)){
                j++;
            }else
                return false;
        }
        if(i!=name.length())
            return false;
        while(j<typed.length()){
            if(name.charAt(name.length()-1)!=typed.charAt(j))
                return false;
            j++;
        }
        return true;
    }
}